---
title: "Quality Control for FMD-GWAS data report- temp"
author: "Siying Huang"
date: "10/11/2016"
output:
  html_document:
    fig_caption: yes
    highlight: haddock
    theme: journal
    toc: yes
    toc_depth: 5
  word_document:
    toc: yes
    toc_depth: '5'
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

### Background 

This document is prepared for the quality control (QC) procedure on genome-wide genotype data typed on Infinium OmniExpressExome-8v1.3 (IOEE8v1.3) and Infinium OmniExpressExome-8v1.4 (IOEE8v1.4) platform for FMD study. As of Oct, 2016, two batches of samples have gone through genotyping. The first round of genotype was generated in Jan, 2016 using platform IOEE8v1.3 and the second round of genotyping was done in Oct, 2016 using platform IOEE8v1.4.   The generic description on IOEE8 array can be found [here](http://www.illumina.com/products/by-type/microarray-kits/infinium-omni-express-exome.html).

The support files can be found on their website for [IOEE8v1.3](http://support.illumina.com/downloads/infinium-omniexpressexome-8-v1-3-product-support-files.html) and for [IOEE8v1.4](https://support.illumina.com/downloads/infinium-omniexpressexome-8-v1-4-support-files.html)
And the product files can be found at : http://support.illumina.com/array/array_kits/infinium_humanomniexpressexome_beadchip_kit/downloads.html

Some useful [tech documents](http://support.illumina.com/array/array_kits/infinium_humanomniexpressexome_beadchip_kit/documentation.html) for IOEE8 plaftform.

```{r,echo=FALSE}
header= c("", "IOEE8v1.3","IOEE8v1.4")
row1= c("No. markers designed on platform", "958497","960919")
row2= c("No. markers returned", "951873","954212")
row3= c("No. markers after QC","925899","933095")
row4= c("Build in Annot file","GRCh37.p13-105","GRCh37.p13-105")
tab= data.frame(rbind(row4,row1,row2,row3))
colnames(tab)= header
library(knitr)
kable(tab,row.names = F,caption = "Summary of two chips")
```

#### Data Jan-16 (1st batch)

The first batch of genotyping data was returned to our lab in Jan 2016. This batch of data contains markers typed on IOEE8v1.3 platform with 958497 markers designed on the platform. A total of 6624 markers were missing in the returned data. The marker distributions from the returned data listed in the table below:

       exm     exm-other    exm-rs        rs    seq_rs       VGS      VGXS 
     235491           14      7539    708013       696       108        12 

Based on the coordinate file from IOEE8v1.3 support file, the chip contains markers:

        exm     exm-IND    exm-other       exm-rs          rs     seq_rs      VGS      VGXS 
    235729          136           14         7647       714145       699      115        12 


Marker ditribution across chromosomes from IOEE8v1.3 chip:  (Chromosome 0- the markers don't have chromosome position on hg 19 assembly)

      0     1    10    11    12    13    14    15    16    17    18    19     2    20    21    22     3 
    699 83020 48003 51672 47332 31771 30669 29579 32782 33010 25183 29676 74080 24594 12910 15411 61096 
        4     5     6     7     8     9    MT     X    XY     Y 
    50050 52854 62629 48538 45316 42528   208 22691   809  1387 


#### Data Oct-16 (2nd batch) 

The second batch of genotyping was generated and returned in Oct 2016. The returned data contains 954212 markers from the IOEE8v1.4 array prior to cleaning. A total of 94 subjects were in Batch 2.

```
      exm exm-other    exm-rs        rs       VGS      VGXS 
   236312        15      7547    710218       108        12 
```

The IOEE8v1.4 array contains marker types:
```
    exm  exm-IND  exm-other      exm-rs           rs       VGS      VGXS 
 236554       134        15        7656       716433       115        12 
```

Marker distribution by chromosomes on the IOEE8v1.4 chip:
```
    0     1    10    11    12    13    14    15    16    17    18    19     2    20    21    22     3 
  707 83251 48065 51835 47420 31866 30776 29652 32917 33126 25195 29797 74237 24661 12954 15456 61166 
    4     5     6     7     8     9    MT     X    XY     Y 
50160 52984 62801 48674 45421 42633   201 22769   803  1392 

```
A total of 6707 probes failed and were missing in the returned data.

#### Multiple markers corresponding to the same rs ID
In both OEE8v1.3 and OEE81.4, there exit multiple markers with different marker names corresponding to the same rs ID. For example, 
```
           Name Chr  MapInfo deCODE.cM.      RsID
     exm1192525  15 99670989   127.5065 rs3743248
      rs3743248  15 99670989   127.5065 rs3743248
 seq1_rs3743248  15 99670989   127.5065 rs3743248
 seq2_rs3743248  15 99670989   127.5065 rs3743248

       Name Chr   MapInfo deCODE.cM.   RsID
  exm121943   1 169513583   171.7348 rs6037
 exm2250216   1 169513583   171.7348 rs6037
     rs6037   1 169513583   171.7348 rs6037
```

There appears to have at least 20161 rsIDs had at least one duplicated markers in OEE8v1.3 and 19974 on OEE8v1.4 platform. Based on the QC result, these duplicated markers seems to tag exact rsIDs:
```
          SNP        GENO O(HET) E(HET)     P CHR N_MISS N_GENO F_MISS CHR A1 A2 MAF NCHROBS	cM BP
seq2_rs3743248 41/275/375 0.398 0.3832 0.3712 15 0 694 0 15 T C 0.2583 1382 127.507 99670989
seq1_rs3743248 41/275/375 0.398 0.3832 0.3712 15 0 694 0 15 T C 0.2583 1382 127.507 99670989
     rs3743248 41/275/375 0.398 0.3832 0.3712 15 0 694 0 15 T C 0.2583 1382 127.507 99670989
```


#### Prepare files for analysis in PLINK

1. [`.ped` file preparation](http://pngu.mgh.harvard.edu/~purcell/plink/data.shtml#ped)
The returned ped file (`Final_dat_DFM.ped`) contains headers and does not have family ID: 

```
Family	Individual	Father	Mother	Sex	Status	rs4477212	rs3094315	rs3131972
          FDP564001	0	0	2	0	A A	  G G	  A A
	      FDP563001	0	0	2	0	A A	  A A	  G G
	      FDP562001	0	0	2	0	A A	  A A	  G G

```
PLINK will read in header and count 6 + nSNP columns (which should be 6+ 2 x nSNPs) and return an error message. Therefore need to remove the header by using command:
```{s}
tail -n +2 Org_dat/Final_dat_DFM.ped > Final_dat_DFM_nh.ped
```
Eventually the `.ped` file should look like:

```
	FDP564001	0	0	2	0	A A	G G	A A
	FDP563001	0	0	2	0	A A	A A	G G
	FDP562001	0	0	2	0	A A	A A	G G

```
2. [`.map` file prepartion](http://pngu.mgh.harvard.edu/~purcell/plink/data.shtml#map)

The `.map` file can be generated from two sources: `success_mark.txt` is returned along with genotype data and contains chromosome and BP position information (build version hg19). This file, however, does not have genetic distance (cM) information. Another way is to download files from IOEE8v1.3 support file `infiniumomniexpressexome-8-v1-3-a-physical-and-genetic-coordinates.zip`. It contains SNP name, chromosome number, BP position, and genetic distance (cM). The `.map` file columns need to follow the order : chr, rs, cm, bp but without column names. It should look like:

```
1	rs4477212	0	82154
1	rs3094315	0	752566
1	rs3131972	0	752721

```
Additionally, non autosomal chromosomes need to be recoded :  
```
     X    X chromosome                    -> 23
     Y    Y chromosome                    -> 24
     XY   Pseudo-autosomal region of X    -> 25
     MT   Mitochondrial                   -> 26
```
3. [`.bed` file preparation](http://pngu.mgh.harvard.edu/~purcell/plink/data.shtml#bed)

Commonly, it will take a while to load `.ped` file up to PLINK. PLINK has the option to reformat `.ped` file into bindary ped file to save time for uploading the data. Simply specify the `.ped` and `.map` files in the directory and use the option `--make-bed`. PLINK will generate the binary ped files named `plink.bed`, map file - ` plink.bim` ,and first six columns - `plink.fam` . Alternatively, you can use `--out` option to specify the output file name.

For analysis using PLINK one can upload data with command:
```{s}
~/plink --bfile filename
```

#### GWAS QC Pipeline  
![](/data-vg-Vol1/Siying/GWAS/GWAS_pipe.png)

### QC Per Individual 
This [GitHub](https://github.com/genepi-freiburg/gwas) website provides all the scripts needed for GWAS QC from [Anderson et al Nature Protocol 2010](http://www.nature.com/nprot/journal/v5/n9/full/nprot.2010.116.html) .

#### 1. Sex check

```{s}
./plink --bfile plink --check-sex 
```
[PLINK](http://pngu.mgh.harvard.edu/~purcell/plink/summary.shtml#sexcheck)  will generate the file `*.sexcheck` . 1= male, 2= female. `F` is  homozygosity rate on X-chromosome. For males it is expected to be 1 and for females it is expected to be less than 0.2. When F value is between 0.2 and 0.8, PLINK will report sex to be inconclusive and assign 0 to the individual.

##### Batch 1

To check which individual has sex labeling error:
```{s}
grep PROBLEM *.sexcheck
```
In this data set, the following individual had sex labeling error:
```
        FDP452001         FDP452001            1            2      PROBLEM      -0.1184
        FDP316001         FDP316001            2            0      PROBLEM       0.4021
        FDP173001         FDP173001            2            0      PROBLEM       0.2976
        FDP134001         FDP134001            1            2      PROBLEM     0.008335
        FDP060001         FDP060001            2            1      PROBLEM       0.9927
  ARCADIA20142663   ARCADIA20142663            2            1      PROBLEM       0.9925
  ARCADIA20132016   ARCADIA20132016            2            0      PROBLEM       0.4076

```

After impose contrain on allle frequency on X chromosome (`--sex-check --maf 0.01`) the following subjects from Batch 1 still have sex mismatch problems:
```
        FDP452001         FDP452001            1            2      PROBLEM     -0.08662
        FDP316001         FDP316001            2            0      PROBLEM       0.4001
        FDP173001         FDP173001            2            0      PROBLEM       0.2986
        FDP134001         FDP134001            1            2      PROBLEM     0.007112
        FDP060001         FDP060001            2            1      PROBLEM        0.996
  ARCADIA20142663   ARCADIA20142663            2            1      PROBLEM       0.9958
  ARCADIA20132016   ARCADIA20132016            2            0      PROBLEM       0.4081

```
##### Batch 2
These subjects don't have sex information in the original file:

```
Family	Individual	Father	Mother	Sex	Status	rs4477212	rs3131972	exm2268640
	ARCADIA20161630	0	0	0	2	A A	G G	C C
	ARCADIA20161629	0	0	0	2	A A	G G	C C
	ARCADIA20161628	0	0	0	2	A A	G G	C C

```

####2. Missing rate per individual

[PLINK](http://pngu.mgh.harvard.edu/~purcell/plink/thresh.shtml#miss2) code: `plink --file mydata --mind 0.1`

**Batch 1**- No individual had call rate < 90%. All subjects retained in the data.   
**Batch 2**- No individual had call rate < 90%. All subjects retained in the data.  

####3. Heterozygosity rate and homozygosity
[PLINK](http://pngu.mgh.harvard.edu/~purcell/plink/ibdibs.shtml#inbreeding) code: `plink --file mydata --het` . PLINK will generate the result to the file `plink.het`. The output looks like:

```
              FID               IID       O(HOM)       E(HOM)        N(NM)            F
        FDP564001         FDP564001       562474    5.595e+05       777536      0.01362
        FDP563001         FDP563001       560316    5.607e+05       779272    -0.001904
        FDP562001         FDP562001       566311    5.617e+05       780696       0.0211
        FDP561001         FDP561001       564782    5.606e+05       779022      0.01918

```

`F` is inbreeding coefficient estimate. [Here](http://www.genetic-genealogy.co.uk/Toc115570144.html) is some background for inbreeding coefficient.


PLINK calculates per individual missing rate and heterozygous rate with options  `--missing` , `--het`. The results are stored in `.imiss` and `.het` files. 
There's a R script (`imiss-vs-het.R`) calculate heterozygous rate using formula : $\frac{N_{NM}-O_{HOM}}{N_{NM}}$. This file requires four arguments: 1. The path directs to where the `.het` and `.imiss` files are stored. 2. The name of the output figure 3. number of standard deviation (SD) to be excluded from the dataset (usually set to 3) and 4. The missingness threshold to exclude the individuals with excessive missing rate (usually set to 3%).

In the shell command, run 

`/usr/bin/Rscript /home/.../imiss-vs-het.R "/data-vg-Vol1/.../filename" "hetplot" "3" "0.03"`

This would produce the figure in below. Each individual is represented by a blue dot. The vertical dash line represents the threshold for missing rate (default is 3%). Any individual passed this threshold should be excluded. The horizontal dash lines represents the 3 SD away from mean heterozygosity rate. Any individual passed these lines should be considered to be removed from the dataset or flagged. Excessive or reduced heterozygosity reate indicate DNA sample contamination or inbreeding respectively.  

**Batch 1**:  
![](/data-vg-Vol1/Siying/GWAS/QC/IND/B1/hetplot.png)  
Note that the average heterozygosity rate is about 27.6% in Batch 1 dataset.
For the following figure use R: `/usr/bin/Rscript /home/siying/plot-het-id.R "/data-vg-Vol1/Siying/GWAS/QC/IND/B2/B2" "/data-vg-Vol1/Siying/GWAS/QC/IND/B2/B2" 
`
![](/data-vg-Vol1/Siying/GWAS/QC/IND/B1/B1het-F.png)

**Batch 2**:

![](/data-vg-Vol1/Siying/GWAS/QC/IND/B2/hetplot.png)
![](/data-vg-Vol1/Siying/GWAS/QC/IND/B2/B2het-F.png)


#### 4. Identification of duplicated and related individuals

##### 4.1 Pruning out markers that are in LD
In [PLINK](http://pngu.mgh.harvard.edu/~purcell/plink/summary.shtml#prune), select markers using VIF pruning routine (default setting), which will generate files `.prune.in` and `.prune.out` Then store the genotype data in the `.bed` file: `--extract B1.prune.in --make-bed`.
```{s}
./plink --bfile /../data --indep 50 5 2 --out B1
./plink --bfile /../data --extract B1.prune.in --make-bed --out B1pruned
```
##### 4.2 Pairwise genome-wide IBD file  
In [PLINK](http://pngu.mgh.harvard.edu/~purcell/plink/ibdibs.shtml), use command `--genome` in the pruned `.bed` files. It will generate a `.genome` file.

```{s}
./plink --bfile B1pruned --genome --out B1
```
Here IBD (kinship coefficient) refers to IBD averaged across genome (see below in `PI_HAT`) , not IBD=2, IBD=1 and IBD=0 at a sinble locus.  

IBD = 1 - duplicates or monozygotic twins  
IBD=0.5 - first degree relatives  
IBD= 0.25 - second degree relatives  
IBD= 0.125 third degree relatives  
IBD= 0.1875 is the cut off for QC.

An example from `B1.genome` output file: these two pairs are unrelated individuals since their coefficients `PI_HAT` are almost 0. Alos, `Z0` almost = 1. 

```
     FID1      IID1       FID2       IID2 RT EZ      Z0      Z1      Z2  PI_HAT PHE       DST  
FDP564001 FDP564001  FDP563001  FDP563001 UN NA  1.0000  0.0000  0.0000  0.0000  -1  0.865210  
FDP564001 FDP564001  FDP562001  FDP562001 UN NA  0.9563  0.0000  0.0437  0.0437  -1  0.872171  

   PPC   RATIO
0.0001  1.7646
0.1494  1.9305
```
     Z0        P(IBD=0)
     Z1        P(IBD=1)
     Z2        P(IBD=2)
     PI_HAT    P(IBD=2)+0.5*P(IBD=1) ( proportion IBD )
     PHE       Pairwise phenotypic code (1,0,-1 = AA, AU and UU pairs)
     DST       IBS distance (IBS2 + 0.5*IBS1) / ( N SNP pairs )
     PPC       IBS binomial test
     RATIO     Of HETHET : IBS 0 SNPs (expected value is 2)

##### 4.3 Identifying related or duplicated subjects 

Remove individual IBD >0.1875; IBD>0.98 identifies duplicates. Some [background](http://csg.sph.umich.edu/abecasis/class/666.14.pdf) for IBD and IBS. 

**Batch 1**:

In shell: `awk '$10 >= 0.185' B1.genome > B1related.txt`
```{s}
wc -l B1related.txt 
83 B1related.txt
``` 
There were 83 pairs shared IBD>0.185, which is equivalent to between 2nd and 3rd degree biological relationships

Similarly, use the shell code :`awk '$10>=0.98 B1.geome >B1duplicatedID.txt'` will generate a file that contains subjects have pairwise kinship coefficient >0.98, which indicates the duplicated samples or identical twins. (Since we only have one pair of MZ twins ( <span style="color:red">FDP004014 and FDP004013 are monozygotic twin brothers.</span>) in our study, the rest of subjects should be duplicated and should be examined and duplicates are to be removed from data.)

```{s}
wc -l B1duplicateID.txt 
36 B1duplicateID.txt
```

In **Batch 1** there were 35 pairs had exact same genome, including the MZ twins. `FDP061001, FDP060001` are supposed to be different subjects. There might be a sample swap for `FDP060001`.  
```{r,echo=FALSE}
dup= read.table("/data-vg-Vol1/Siying/GWAS/QC/IND/B1/B1duplicateID.txt",header = T)
dup[,c("IID1","IID2","Z0","Z1","Z2","PI_HAT")]
```

*1st degree relatives in Batch 1*:

`awk '$10 >0.4 && $10 <0.8' B1related.txt >B1fams`

```{r,echo=FALSE}
fam= read.table("/data-vg-Vol1/Siying/GWAS/QC/IND/B1/B1fams")
colnames(fam)= c("FID1","IID1","FID2","IID2", "RT","EZ","Z0","Z1","Z2","PI_HAT", "PHE","DST","PPC","RATIO")
fam[,c("IID1","IID2","Z0","Z1","Z2","PI_HAT")]
```

Download the `run-IBD-QC.pl` file from: https://github.com/genepi-freiburg/gwas/blob/master/cleaning-pipeline/aux/run-IBD-QC.pl  
In shell, run: `perl /home/siying/run-IBD-QC.pl  B1` (requires installation of Perl). This command will scan the `.imiss` file  and output individuals with lowest call rate to file `fail-IBD-QC.txt` for subsequent removal. The `fail-IBD-QC.txt` output for **batch 1** (first four lines):

```
FDP563001 FDP563001
ARCADIA20140700 ARCADIA20140700
FDP557001 FDP557001
FDP555001 FDP555001
```
55 subjects in `fail-IBD-QC.txt`  

Use `R` script (`plot-IBD.R`) to visualize the IBD result from `B1.genome` file

Pairwise kinship coefficient among study subjects  
![](/data-vg-Vol1/Siying/GWAS/QC/IND/B1/pairwise-IBD-graph.png)  
X-axis represent the individuals in the study, y axis represent the average IBD. Red line represent the QC cut off for related individual at IBD=0.185.

Histogram of mean pairwise IBD  
![](/data-vg-Vol1/Siying/GWAS/QC/IND/B1/pairwise-IBD-histogram.png)
Average *IBS* distance  
![](/data-vg-Vol1/Siying/GWAS/QC/IND/B1/DST-plot.png)         

IBS distance = $\frac{IBS2 + 0.5*IBS1}{N_{SNP pairs}}$

![](/data-vg-Vol1/Siying/GWAS/QC/IND/B1/pairwiseIBD.png)  
This plot compares IBD1 against IBD0. The most of unrelated subjects should have more non-shared allels (Z0)  than one shared alleles (Z1), such as in the right lower conner. In the left lower conner, several subjects had Z0=0, Z1=0 indicating duplicated samples or MZ twins (Z2=1). In the left upper conner pairs had Z0=0 Z1=1, indicating parent-offspring relationship, whereas pairs had Z1 = 0.5, Z0 =0.25 are siblings. And so on, one can infer third and even further degrees of relateness in the plot.   


**Batch 2**: 

` awk '$10>0.1875' B2.genome `  
Three individuals are likely to be 1st degree relatives (parent -offspring) and likely to be a trio.  

```
             FID1             IID1             FID2             IID2 RT    EZ
  ARCADIA20161501  ARCADIA20161501  ARCADIA20161485  ARCADIA20161485 UN    NA
  ARCADIA20161495  ARCADIA20161495  ARCADIA20161485  ARCADIA20161485 UN    NA

      Z0      Z1      Z2  PI_HAT PHE       DST     PPC   RATIO
  0.0015  0.9737  0.0248  0.5116  -1  0.899201  1.0000 1879.0000
  0.0026  0.9854  0.0120  0.5047  -1  0.897811  1.0000 3729.0000
```

![](/data-vg-Vol1/Siying/GWAS/QC/IND/B2/pairwise-IBD-graph.png)  
![](/data-vg-Vol1/Siying/GWAS/QC/IND/B2/pairwiseIBD.png)  
![](/data-vg-Vol1/Siying/GWAS/QC/IND/B2/pairwise-IBD-histogram.png)  

**All study subjects**

I merged study subjects from Batch 1 and Batch 2. Results are consistent with the previous analysis except that one subject from Batch 1 (`FDP543001`) is duplicated from Batch 2 (`ARCADIA20161479`). The code to detect this is in `QC.R` 

![](/data-vg-Vol1/Siying/GWAS/QC/IND/ALL/IBD/pairwise-IBD-graph.png)  
![](/data-vg-Vol1/Siying/GWAS/QC/IND/ALL/IBD/pairwiseIBD.png)  
![](/data-vg-Vol1/Siying/GWAS/QC/IND/ALL/IBD/pairwise-IBD-histogram.png)  


#### 5. Population structure 
##### 5.1 IBS similarity matrix
In [PLINK](http://pngu.mgh.harvard.edu/~purcell/plink/strat.shtml) using the pruned bed files, use command `--cluster --matrix`. PLINK will generate file `.mibs`, which contains a symmetric matrix of IBS distance for all pairs of individuals. Alternatively, use command `--cluster --distance-matrix` to generate the file `.mdist`.  
In R, plot the individual IBS distance for visual inspection:

**Batch 1**:

```{r,echo=FALSE,fig.height=6, fig.width=6}
ibsd= read.table("/data-vg-Vol1/Siying/GWAS/QC/IND/B1/plink.mds",header = T)
par(mfrow=c(2,2),pch=20, col=rgb(0.1,0.1,0.1,0.3),oma = c(0, 0, 2, 0), bty="l", font.main=1)
plot(ibsd$C1 ~ibsd$C2)
plot(ibsd$C1 ~ibsd$C3)
plot(ibsd$C3 ~ibsd$C2)
plot(ibsd$C3 ~ibsd$C4)
mtext(text = "Batch 1: Jan-16",outer = T,side = 3,line=-21,col="blue")
```


**Batch 2**:

```{r,echo=FALSE,fig.height=6, fig.width=6}
mds= read.table("/data-vg-Vol1/Siying/GWAS/QC/IND/B2/B2IBS.mds", header = T)
par(mfrow=c(2,2),pch=20 ,col= rgb(0.1,0.1,0.1,0.3),oma = c(0, 0, 2, 0), bty="l", font.main=1)
plot(mds$C1 ~mds$C2)
plot(mds$C1 ~mds$C3)
plot(mds$C3 ~mds$C2)
plot(mds$C3 ~mds$C4)
mtext(text = "Batch 2: Oct-16",outer = T,side = 3,line=-21,col="blue")
```

##### 5.2 PCA

Population structure can be assessed using PCA approach using ancessetry informative markers (AIMs) across genome. The requirement for the number of this kind of markers varies from ~20 to hundreds. However, if not sure about AIMs, about 100 K markers across genoome is recommended.  This procedure takes very long time in PLINK. Alternatively one can use [`smartpca`](https://www.hsph.harvard.edu/alkes-price/eigensoft-frequently-asked-questions/) to reduce computational time. In order to inspect the population distributions, one need to use a reference that contains major ancestry distributions. In this QC, I use [HapMap 3](http://www.sanger.ac.uk/resources/downloads/human/hapmap3.html) reference as Anderson et al recommended (see ref above). The HapMap 3 reference can be downloaded from [here](https://github.com/genepi-freiburg/gwas/find/master). `hapmap3r2_CEU.CHB.JPT.YRI.no-at-cg-snps.txt ` provide a list of 1047171 markers that are available in 395 HapMap3 subjects. These subjects are from CEU, CHB, JPT and YRI ancestral populations. These reference populations are sufficient to describe major ancestral populations.

To compare with the reference genome, we need to trim the markers that matched up with HapMap3 reference markers. In PLINK, make a sub-dataset contains the markers from HapMap3 reference. 

```
./plink --bfile /filename --extract /.../hapmap3r2_CEU.CHB.JPT.YRI.no-at-cg-snps.txt --make-bed --out B1hapmap
```

Next, merge HapMap reference data with study data. 
```
./plink --bfile B1hapmap --bmerge hapmap3r2_CEU.CHB.JPT.YRI.founders.no-at-cg-snps.bed hapmap3r2_CEU.CHB.JPT.YRI.founders.no-at-cg-snps.bim 
hapmap3r2_CEU.CHB.JPT.YRI.founders.no-at-cg-snps.fam  
--extract B1.prune.in --make-bed --out B1hapmap3r2_pruned
```

PLINK will generate error message because SNPs are discordant between the two datasets (e.g. A->T and C->G) :

```
# Found 237 SNPs that do not match in terms of allele codes
# Might include strand flips, although flipped A/T and C/G SNPs will be undetected)
# Writing problem SNPs to [ B1hapmap3r2_pruned.missnp ]
# ERROR: Stopping due to mis-matching SNPs -- check +/- strand?
```
And PLINK will generate a `.missnp` file- list of markers that need to be fliped. 

##### 5.2.1 Flipe strand   
To solve this issue, one needs to flip the strand in these markers listed in the `.missnp` file using [PLINK](http://pngu.mgh.harvard.edu/~purcell/plink/dataman.shtml#flip):
```{s}
./plink --bfile B1hapmap --flip B1hapmap3r2_pruned.missnp --recode --out B1hapmap3r2_pruned_fliped
```
The `--recode` option will generate a new dataset in `.ped` and `.map` file. One can directly generate binary ped files using `--recode --make-bed` . But I prefer to generate `.ped` file to check if the datasets are correctly merged later. 
After the this step, the flipped dataset can be used to merge with HapMap3 reference.

##### 5.2.2 Merge with HapMap 3 reference  
There are two options to merge files: `--merge` merges two `.ped` files and `--bmerge` merges two binary ped files . Since the HapMap3 files are in binary file format, I first converted `.ped` dataset to bindary ped file then `--bmerge` two datasets using command:
```{s}
/plink --bfile B1hapmap3r2_pruned_fliped --bmerge hapmap3r2_CEU.CHB.JPT.YRI.founders.no-at-cg-snps.bed hapmap3r2_CEU.CHB.JPT.YRI.founders.no-at-cg-snps.bim hapmap3r2_CEU.CHB.JPT.YRI.founders.no-at-cg-snps.fam  --extract B1.prune.in --recode --out B1hapmap3r2_pruned_test
```
You should see this in the log:

```
...
# 1047171 markers to be merged from [ /home/siying/hapmap3r2_CEU.CHB.JPT.YRI.founders.no-at-cg-snps.bim ]
# Of these, 592859 are new, 454312 already exist in current data
# 395 individuals merged from [ /home/siying/hapmap3r2_CEU.CHB.JPT.YRI.founders.no-at-cg-snps.fam ] 
# Of these, 395 were new, 0 were already in current data
...
# 1086 founders and 3 non-founders found
...
# Writing recoded ped file to [ B1hapmap3r2_pruned_test.ped ] 
# Writing new map file to [ B1hapmap3r2_pruned_test.map ] 
```
There should be 1089 subjects in the merged dataset (649 FMD +395 HapMap3)

To check if two datasets are correctly merged, use the shell command: `wc -l B1hapmap3r2_pruned_test.ped`   It will give the number of lines (# of subjects) in the dataset.


##### 5.2.3 PCA analysis using smartpca  

Next, two files need to be prepared for the PCA analysis in the `smartpca` program (Some data preparation requirement can be found in [housekeeping](http://fibro:8787/files/housekeeping.html) material.):  
`.pedsnp`:  
`cp B1hapmap3r2_pruned.bim B1hapmap3r2_pruned.pedsnp`  
`.pedind`:  
`cp B1hapmap3r2_pruned.fam B1hapmap3r2_pruned.pedind`  
The `pedind` need to specify the ancestry population or, in the study dataset, the "Case" and "Control". This can be edited in `R`.   

```{r,eval=FALSE}
plink= read.table("/data-vg-Vol1/Siying//GWAS/QC/IND/B1/test/plink.fam",header = F)
race= ifelse(plink$V6 == -9, "Case",plink$V6)
plink$V6=race
write.table(plink,"/data-vg-Vol1/Siying//GWAS/QC/IND/B1/test/plink.fam",row.names = F, quote = F,  col.names = F )

```

Or use `awk '{if($6=="2") $6="Case"; print$0}' B2_pca.fam >B2_pca.pedind` under linux. 

`smartpca` can read in `.bed` file as it it. So first, one needs to convert `.ped` to `.bed` file (In my case, output were `plink.*`). Then use the `perl` wrapper to run `smartpca`:

```{s}
perl smartpca.perl -i plink.bed -a plink.pedsnp -b plink.pedind -o test.pca -p test.plot -e test.eval -l test.log -k 2 -t 2 
```
For the details of these options in command please see the `README` file for the `smartpca`.

Visualizing population structure using R script `plot-pca-results.R`. I adapted the code to make it more scaled to our data. `/usr/bin/Rscript plot-pca-results.R "/path.../test.pca.evec" "PCA.png"`

**Batch 1** result:   
![](/data-vg-Vol1/Siying/GWAS/QC/IND/B1/test/PCA.png)

**Batch 2** result:   
![](/data-vg-Vol1/Siying/GWAS/QC/IND/B2/PCA.png)

**All study subjects**  
![](/data-vg-Vol1/Siying/GWAS/QC/IND/ALL/test/PCA-ALL.png)


### QC per marker
Because the call rate for all individuals maintained above 90%, this per markers QC is done in all individual prior to the triming the duplicated and related subjects. The same procedure will be performed again once the subjects are cleaned.

#### 1. Missing rate
##### 1.1 Overall missing rate (markers and individuals)
Use [PLINK](http://pngu.mgh.harvard.edu/~purcell/plink/summary.shtml#missing) command :  
`plink --file data --missing`
PLINK will generate two files: `*.imiss` missing rate per individuals and `*.lmiss` missing rate per SNP. For the details of the output format, check PLINK website.

Output example for per individual missing rate in Batch 1:
```
              FID               IID MISS_PHENO   N_MISS   N_GENO   F_MISS
        FDP564001         FDP564001          N     4485   950495 0.004719
        FDP563001         FDP563001          N     2091   950495   0.0022
        FDP562001         FDP562001          N      504   950495 0.0005303
```
Check the subjects who has missing rate > 1% (cloumn 6 -Proportion of missing SNPs ) using command:
` awk '$6 > 0.01' B1.imiss`.  
Two subjects had more than 1% missing in **Batch 1**. But the missing rate was not very high so can be included in the analysis.
None in **Batch 2** had missing rate > 1%.


Example output for per SNP missing rate in **Batch 1**.

```
 CHR              SNP   N_MISS   N_GENO   F_MISS
   1        rs6664362      428      694   0.6167
   1         exm27104      387      694   0.5576
   1         exm54432      431      694    0.621
   1       rs12081621      403      694   0.5807
```
Check the SNPs has missing rate >3% (Column 5 in `lmiss` file):

```{s}
awk '$5 > 0.03' B1.lmiss| less  
awk '$5 > 0.03' B1.lmiss > B1lowcallSNP.txt  
wc -l B1lowcallSNP.txt 
9179 B1lowcallSNP.txt
```

A total of 9178 SNPs had call rate <97%.
Similarly, in **Batch 2** 3198 markers had call rate <97%.(<span style="color:red">These markers are to be excluded from the analyses</span>) 

```{s}
wc -l B2lowcallSNP.txt 
3198 B2lowcallSNP.txt
```


```{r, echo=FALSE}
miss= read.table("/data-vg-Vol1/Siying/GWAS/QC/IND/B1/B1lowcallSNP.txt", header = T)
miss2= read.table("/data-vg-Vol1/Siying/GWAS/QC/IND/B2/B2lowcallSNP.txt",header = T)
par(mfrow=c(2,1),font.main=1,mar=c(4,3,2,1), cex.main=0.8)
hist(miss$F_MISS,breaks = length(miss$F_MISS)/100, col=rgb(0.2,0.1,0.2,0.3), main="Missing call rate Data Jan -16", xlab = "Missing rate in markers call missing rate > 3%- Batch 1 ")
hist(miss2$F_MISS,breaks = length(miss2$F_MISS)/50, col=rgb(0.2,0.1,0.2,0.3), main="Missing call rate  Data Oct -16", xlab = "Missing rate in markers with call missing rate > 3%- Batch 2 ")

```

##### 1.2 Differential missing 
This step is an extra. The purpose for this procedure is to check if the missing pattern of SNP calls is related to any factors, e.g. MAF and chromosomes. To check the pattern, one needs to merge the output files from PLINK `*.lmiss` and `*.frq` using Unix command:  
`join -1 2 -2 2 -a 1 -a 2 B1.lmiss plink.frq >  maf_miss.merged`. Since the two output files used exact same order of SNPs, we don't need to sort them at the first place. Otherwise, one needs to sort out the files with regard to SNPs using the code `sort file` in Unix. Then we can use R to visually check the results using the code below.  

```{r, eval=FALSE}
data= read.table("/data-vg-Vol1/Siying/GWAS/QC/IND/B1/maf_miss.merged",header = T)
png("/data-vg-Vol1/Siying/GWAS/QC/IND/B1/callratecheck.png", width= 600, height=800)
par(mfrow=c(2,1),mar=c(4,4,1,1),cex.lab=0.9, cex.axis= 0.8)
plot(data$MAF , data$F_MISS , pch= 20,col= rgb(0,0.2,1,0.3), xlab= "MAF", ylab="SNP missing rate")
abline(v= 0.05,col="red"); abline(h= 0.03, col="red")
plot(data$CHR , data$F_MISS , pch= 20,col= rgb(0.3,0.2,1,0.3), xlab= "Chromosome", ylab="SNP missing rate")
abline(h=0.03, col="red")
dev.off()
```

It will produce two plots visualizing SNP call missing rate with regard to minor allele frequency and chromosomes:  
 **Batch 1**: 
![](/data-vg-Vol1/Siying/GWAS/QC/IND/B1/callratecheck.png)


In the upper panel, the horizontal red line represents SNP missing rate at 3% and the vertical red line represent MAF =0.05. In the lower panel, the red horizontal line represents SNP missing rate at 3%.

 **Batch 2**: 

```{r, eval=FALSE, echo=FALSE}
data= read.table("/data-vg-Vol1/Siying/GWAS/QC/IND/B2/B2_maf_frq_hwe_coord",header = T)
png("/data-vg-Vol1/Siying/GWAS/QC/IND/B2/callratecheck.png", width= 600, height=800)
par(mfrow=c(2,1),mar=c(4,4,1,1),cex.lab=0.9, cex.axis= 0.8)
plot(data$MAF , data$F_MISS , pch= 20,col= rgb(0,0.2,1,0.3), xlab= "MAF", ylab="SNP missing rate")
abline(v= 0.05,col="red"); abline(h= 0.03, col="red")
plot(data$CHR , data$F_MISS , pch= 20,col= rgb(0.3,0.2,1,0.3), xlab= "Chromosome", ylab="SNP missing rate")
abline(h=0.03, col="red")
dev.off()
```

![](/data-vg-Vol1/Siying/GWAS/QC/IND/B2/callratecheck.png)

#### 2. Mnor allele frequency (MAF)  
Using code: `--freq`, PLINK will generate file `*.frq` 

Extracting the the MAF less than 0.05: `awk '$5 <0.05' plink.frq > B1MAF.txt`. 
Creating a file to remove the low MAF: `awk '{print $2}' B1MAF.txt >MAF.prune.out`.  
328130 markers had MAF <0.05 and their distributions by marker types are given below:

**Batch 1:**
```{r}
maf_out= read.table("/data-vg-Vol1/Siying/GWAS/QC/IND/B1/MAF.prune.out")
markertype= gsub("([[:alpha:]]+)([[:digit:]]+)","\\1", maf_out[,1])
table(markertype)
```

**Batch 2:**
```{r}
maf_out= read.table("/data-vg-Vol1/Siying/GWAS/QC/IND/B2/B2MAFpruneout")
markertype= gsub("([[:alpha:]]+)([[:digit:]]+)","\\1", maf_out[,1])
table(markertype)
```

#### 3. Hardy- Weinberg Equilibrium  
In PLINK, use option `--hardy` `--hwe 0.001`. PLINK will perform HWE test in cases, controls and all subjects separately. The output `*.hwe` contains test results in these three groups. Since we are testing all FMD cases, just simple use `ALL` group and in shell command: `grep ALL B1HWE.hwe > B1HWErest`. It will extract the rest results for 951873 markers in all FMD cases.    

Output sample from `B1HWErest`: 

```
# CHR            SNP     TEST   A1   A2                 GENO   O(HET)   E(HET)            P 
# 1        rs4477212      ALL    0    A              0/0/691        0        0            1
# 1        rs3094315      ALL    G    A           26/233/432   0.3372   0.3274       0.4865
# 1        rs3131972      ALL    A    G           29/239/423   0.3459   0.3374       0.5735
# 1       exm2268640      ALL    T    C             0/14/677  0.02026  0.02006            1

```
Using shell command: `awk '$9 <0.001' B1HWErest > B1HWEflag` for **Batch 1** to identify the markers with HWE test p-value <0.001.

```{s}
wc -l B1HWEflag
15832 B1HWEflag
```
There are 15832 markers failed HWE test at p<0.001.

For **Batch 2**, 750 markers had HWE test p-value <0.001. (However, note that there were only 94 subjects in Batch 2, so the tests results may be interpreted with cautions.)  

```{s}
wc -l B2HWEflag 
750 B2HWEflag
```

The following R code visualize the HWE test ressults with regard to the genotyping quality (e.g. missing rate, MAF). This is done in the file merged `*.lmiss`, `*.frq`, and `*.hwe`.

**Batch 1**

```{r,eval=FALSE}
data= read.table("/data-vg-Vol1/Siying/GWAS/QC/IND/B1/maf_miss_hwe",header = T)

png("/data-vg-Vol1/Siying/GWAS/QC/IND/B1/hwe_maf.png")
plot(data[,"MAF"], data[,"P"], ylab="p-value", xlab="MAF", col= rgb(0,0.2, 0.3, 0.3), pch=20)
title("HWE test p-value by MAF")
abline(v=0.05, col="red")
dev.off()

png("/data-vg-Vol1/Siying/GWAS/QC/IND/B1/hwe_miss.png")
plot(data[,"F_MISS"], -log10(data[,"P"]) ,ylab="-log10(P)", xlab="SNP missing rate", col= rgb(0,0.2, 0.3, 0.3), pch=20)
abline(v=0.03, col="blue")
abline(h= 3, col="red",lty=2)
title("HWE test p-value by SNP missing rate")
dev.off()

```

![](/data-vg-Vol1/Siying/GWAS/QC/IND/B1/hwe_miss.png)  
The blue vertical line indicate the SNP missing rate at 3%. The red dashed line indicate the HWE test p value at 0.001. 

![](/data-vg-Vol1/Siying/GWAS/QC/IND/B1/hwe_maf.png)  
The red vertical line marks the MAF =0.05.


**Bacth 2 **
```{r,eval=FALSE, echo=FALSE}
data= read.table("/data-vg-Vol1/Siying/GWAS/QC/IND/B2/B2_maf_frq_hwe_coord",header = T)

png("/data-vg-Vol1/Siying/GWAS/QC/IND/B2/hwe_maf.png")
plot(data[,"MAF"], data[,"HWE"], ylab="p-value", xlab="MAF", col= rgb(0,0.2, 0.3, 0.3), pch=20)
title("HWE test p-value by MAF")
abline(v=0.05, col="red")
dev.off()

png("/data-vg-Vol1/Siying/GWAS/QC/IND/B2/hwe_miss.png")
plot(data[,"F_MISS"], -log10(data[,"HWE"]) ,ylab="-log10(P)", xlab="SNP missing rate", col= rgb(0,0.2, 0.3, 0.3), pch=20)
abline(v=0.03, col="blue")
abline(h= 3, col="red",lty=2)
title("HWE test p-value by SNP missing rate")
dev.off()
```


![](/data-vg-Vol1/Siying/GWAS/QC/IND/B2/hwe_miss.png)  
The blue vertical line indicate the SNP missing rate at 3%. The red dashed line indicate the HWE test p value at 0.001. 

![](/data-vg-Vol1/Siying/GWAS/QC/IND/B2/hwe_maf.png)  
The red vertical line marks the MAF =0.05.

#### 4. Miscellaneous

In PLINK log: 
```
...
27 SNPs with no founder genotypes observed
691 founders and 3 non-founders found
36035 heterozygous haploid genotypes; set to missing
Writing list of heterozygous haploid genotypes to [ plink.hh ]
...
```
##### 4.1. The 27 SNPs with no founder genotpyes observed: 
```{r, eval=FALSE}
nof= read.table("/data-vg-Vol1/Siying/GWAS/QC/IND/B1/plink.nof")
mafmisshwe=read.table("/data-vg-Vol1/Siying/GWAS/QC/IND/B1/maf_miss_hwe",header = T)
ind= mafmisshwe$SNP %in% nof[,1]
rownames(mafmisshwe)= mafmisshwe$SNP
mafmisshwe[ind,]
```

```
                  SNP  GENO O.HET. E.HET. P CHR N_MISS N_GENO F_MISS CHR.1 A1 A2 MAF NCHROBS
VG01S1133   VG01S1133 0/0/0    NaN    NaN 1   1    694    694      1     1  0  0  NA       0
VG01S3104   VG01S3104 0/0/0    NaN    NaN 1   1    694    694      1     1  0  0  NA       0
rs2241199   rs2241199 0/0/0    NaN    NaN 1   2    694    694      1     2  0  0  NA       0
rs2296197   rs2296197 0/0/0    NaN    NaN 1   6    694    694      1     6  0  0  NA       0
rs2069525   rs2069525 0/0/0    NaN    NaN 1  15    694    694      1    15  0  0  NA       0
exm2252654 exm2252654 0/0/0    NaN    NaN 1  16    694    694      1    16  0  0  NA       0
rs764688     rs764688 0/0/0    NaN    NaN 1  17    694    694      1    17  0  0  NA       0
rs1807466   rs1807466 0/0/0    NaN    NaN 1  22    694    694      1    22  0  0  NA       0
VGXS34742   VGXS34742 0/0/0    NaN    NaN 1  23    694    694      1    23  0  0  NA       0
VGXS34722   VGXS34722 0/0/0    NaN    NaN 1  23    694    694      1    23  0  0  NA       0
VGXS34711   VGXS34711 0/0/0    NaN    NaN 1  23    694    694      1    23  0  0  NA       0
exm2263283 exm2263283 0/0/0    NaN    NaN 1  24    133    133      1    24  0  0  NA       0
exm2263287 exm2263287 0/0/0    NaN    NaN 1  24    133    133      1    24  0  0  NA       0
exm2263291 exm2263291 0/0/0    NaN    NaN 1  24    133    133      1    24  0  0  NA       0
exm2263294 exm2263294 0/0/0    NaN    NaN 1  24    133    133      1    24  0  0  NA       0
exm2263295 exm2263295 0/0/0    NaN    NaN 1  24    133    133      1    24  0  0  NA       0
exm2263298 exm2263298 0/0/0    NaN    NaN 1  24    133    133      1    24  0  0  NA       0
exm2263214 exm2263214 0/0/0    NaN    NaN 1  24    133    133      1    24  0  0  NA       0
exm2263217 exm2263217 0/0/0    NaN    NaN 1  24    133    133      1    24  0  0  NA       0
exm2263221 exm2263221 0/0/0    NaN    NaN 1  24    133    133      1    24  0  0  NA       0
exm2263228 exm2263228 0/0/0    NaN    NaN 1  24    133    133      1    24  0  0  NA       0
exm2263231 exm2263231 0/0/0    NaN    NaN 1  24    133    133      1    24  0  0  NA       0
exm2263234 exm2263234 0/0/0    NaN    NaN 1  24    133    133      1    24  0  0  NA       0
exm2263237 exm2263237 0/0/0    NaN    NaN 1  24    133    133      1    24  0  0  NA       0
exm2263238 exm2263238 0/0/0    NaN    NaN 1  24    133    133      1    24  0  0  NA       0
exm2263239 exm2263239 0/0/0    NaN    NaN 1  24    133    133      1    24  0  0  NA       0
exm2263242 exm2263242 0/0/0    NaN    NaN 1  24    133    133      1    24  0  0  NA       0
```

##### 4.2 Non-founder  

`awk '{if($3!=0) print $0}' plink_cp.fam `

```
FDP018012 FDP018012 FDP018004 Dxx000000000001 1 -9
FDP018009 FDP018009 Dxx000000000002 FDP018001 1 -9
FDP018007 FDP018007 Dxx000000000003 FDP018001 1 -9
```
The raw `.fam` file was not cleaned the 3rd and the 4th columns should have values of 0. The pedigree informtion is available on the housekeeping page. There are three pedigrees in this dataset (`FDP018xxx`, `FDP004xxx`, `FDP0143xxx`). The `.fam` file needs to be updated with pedigree information.   

##### 4.3 Heterozygous haploid

''Normally, heterozygous haploid and nonmale Y chromosome genotype calls are logged to plink.hh and treated as missing by all analysis commands, but left undisturbed by --make-bed and --recode (since, once gender and/or chromosome code errors have been fixed, the calls are often valid).''

`head plink.hh`

```
FDP134001	FDP134001	rs5939319
FDP134001	FDP134001	exm1625831
ARCADIA20111405	ARCADIA20111405	rs1419931
FDP550001	FDP550001	rs3671
FDP452001	FDP452001	rs3671
FDP390001	FDP390001	rs3671
FDP362001	FDP362001	rs3671
FDP326001	FDP326001	rs3671
FDP286001	FDP286001	rs3671
```
```{r,eval= TRUE}
hh= read.table("/data-vg-Vol1/Siying/GWAS/QC/IND/B1/plink.hh")
# Check number of subjects being flagged
length(unique(hh$V1))
# Check number of SNPs being reported
length(unique(hh$V3))
# Check the marker ditributions on the chromosomes
hh.snp= as.character(unique(hh$V3))
coord13= read.table(unz("/data-vg-Vol1/Siying/GWAS/OEE1-3_supportfile/infiniumomniexpressexome-8-v1-3-a-physical-and-genetic-coordinates.zip","InfiniumOmniExpressExome-8v1-3_A_Physical-and-Genetic-Coordinates.txt"),sep = '\t' ,header = T)
ind= coord13$Name %in% hh.snp
rownames(coord13)= coord13$Name
hh.chr= coord13[ind,"Chr"]
table(hh.chr)

fam = read.table("/data-vg-Vol1/Siying/GWAS/bed/plink.fam")
head(fam)
rownames(fam)= fam$V1
hh.sex= fam[as.character(unique(hh$V1)),5]
table(hh.sex)
```

#### 5 Data cleaning  

A total of 638 subjects retained in Batch 1 and 93 subjects retained in Batch 2.

##### 5.1 Remove duplications  

Sample duplications, genetic duplications (MZ twins) and sample swap will be removed from dataset. There are  duplicated 35 samples in Batch 1 (including MZ twins - `FDP004014` removed) , 1 sample in Batch 2 duplicated in Batch 1 (see [4.3 Identifying related or duplicated subjects](#identifying-related-or-duplicated-subjects) for details). Duplicated samples with lower call rates will be removed from the dataset. A total of 35 IDs to be exluded are stored in `Duplicated-rm.txt` after running `rm-duplicate.pl` in `perl` in **Batch 1** and 0 in **Batch 2**. The duplication pair between Batch 1 and Batch 2 are maintained in the data for now. 

##### 5.2 Pedigree membership and sex mismatch

*Pedigree structure*  
There were 16 non-founders in Batch 1 (original), xx non-founders from ARCADIA and one non-founder in Batch 2. All the non-founders were removed from the dataset.

**Batch 1**  
There were three pedigrees from previous family study (FDP018, FDP004, and FDP143) in Batch 1 (details in [housekeeping material](http://fibro:8787/files/housekeeping.html)). I have changed the `FID` and `PID` and `MID` according to the pedigrees in `QC.R` . Below is the updated information before removal of duplicated samples:  

```
       FID       IID       PID       MID sex pheno
269 FDP143 FDP143004         0 FDP134002   1     2
270 FDP143 FDP143002         0         0   2     2
271 FDP143 FDP143001         0         0   2     2
354 FDP018 FDP018018         0 FDP018005   2     2
355 FDP018 FDP018016         0 FDP018005   1     2
356 FDP018 FDP018014 FDP018003         0   1     2
357 FDP018 FDP018013 FDP018003         0   2     2
358 FDP018 FDP018004 FDP018010 FDP018011   1     2
359 FDP018 FDP018012 FDP018004         0   1     2
360 FDP018 FDP018001 FDP018010 FDP018011   2     2
361 FDP018 FDP018009         0 FDP018001   1     2
362 FDP018 FDP018007         0 FDP018001   1     2
371 FDP004 FDP004014         0 FDP004003   1     2
372 FDP004 FDP004013         0 FDP004003   1     2
373 FDP004 FDP004012         0 FDP004003   2     2
374 FDP004 FDP004008         0         0   2     2
375 FDP004 FDP004004         0         0   2     2
376 FDP004 FDP004002         0         0   2     2
377 FDP004 FDP004001         0         0   2     2
```

Additionally, a total of 7 ARCADIA subjects were excluded from the study due to the relateness with other ARCADIA study subjects and based on their call rates (Please refer to the file `fail-IBD-QC.txt` from[4.3 Identifying related or duplicated subjects](#identifying-related-or-duplicated-subjects):  

```
            FID             IID
ARCADIA20132455 ARCADIA20132455 
ARCADIA20112085 ARCADIA20112085 
ARCADIA20101840 ARCADIA20101840 
ARCADIA20122639 ARCADIA20122639 
ARCADIA20121582 ARCADIA20121582 
ARCADIA20102031 ARCADIA20102031 
ARCADIA201000976 ARCADIA201000976
```

Finally there's one sample duplicated in Batch 2 (`FDP543001, ARCADIA20161479`). Since the platform of Batch 2 covered more markers than that of Batch 1, the sample in Batch 1 is going to be removed. 

**Bactch 2**  
In Batch 2, there was an suspicious trio pedigree (see section [4.3 Identifying related or duplicated subjects](#identifying-related-or-duplicated-subjects)). The updated `B2.fam` information:   

```
               FID             IID             PID             MID sex pheno
37 ARCADIA20161485 ARCADIA20161501               0               0   2     2
43 ARCADIA20161485 ARCADIA20161495               0               0   1     2
53 ARCADIA20161485 ARCADIA20161485 ARCADIA20161495 ARCADIA20161501   1     2
```

*Sex mismatch*  

**Batch 1**  
There were 7 subjects with problematic sex info (see [Sex check](#sex-check) for details). Upon confirmation from the original data, some of the sex information in `FDPxxx` subjects were updated. Subject `FDP060001` was a mislabeled sample, which has exact same DNA as `FDP061001`, therefore were removed from this data; hence the sex mismatch was not corrected. As for `ARCADIA20132016`, clinical data confirm the female sex, so the original sex will be retained in spite of discordant genetic sex info. Sex for subject `ARCADIA20142663` is corrected to genetic sex (clinic sex female). Below are the subjects with updated sex info.  

```
                FID             IID PID MID sex pheno
40        FDP452001       FDP452001   0   0   2     2
142       FDP316001       FDP316001   0   0   2     2
245       FDP173001       FDP173001   0   0   2     2
278       FDP134001       FDP134001   0   0   2     2
390 ARCADIA20142663 ARCADIA20142663   0   0   1     2
```
**Batch 2**  
There was no sex information in the original Batch 2 data. So the genetic sex is assigned to the data. There were 13 males, 81 females in Batch 2 (genetic info).

##### 5.3 Other concerns in samples  
Subjects with diverse ancestry information were retained in the datasets.These issues will be addressed in the data analysis steps.  

##### 5.4 Remove markers with low call rate and low minor allele count (MAC)   

In both batches markers with <99% call rate and MAC <5 were removed from the final dataset. Upon removal 925899 markes were maintained from 638 subjects in Batch 1 and 954212 markers in 93 subjects in Batch 2.  
The cleaned data is temporarily stored in bed file `~/GWAS/QC/IND/tmp/B1_clean.*`, `~/GWAS/QC/IND/tmp/B2_clean.*`

**Batch 1**  
After cleaning the marker type distributions:    

```
markertype
      exm exm-other    exm-rs        rs    seq_rs       VGS      VGXS 
   229598        14      7279    688214       682       103         9 
```
; by chromosome:

```
    1     2     3     4     5     6     7     8     9    10    11    12    13    14    15    16    17 
80572 71928 59236 48411 51113 60359 46908 43893 41218 46541 49918 45876 30718 29758 28622 31649 31942 
   18    19    20    21    22    23    24    26 
24285 28531 23878 12447 14911 21708  1312   165 
```
**Note that PLINK doesn't clean non-autosomal chromosomes**
```{r,eval=FALSE}
table(b1miss[b1miss$F_MISS>=0.01,"CHR"]) 
# 23  24  26 
# 64 151   1 
```

**Batch 2**
```
markertype
      exm exm-other    exm-rs        rs       VGS      VGXS 
   231597        15      7332    694036       106         9 
```
Marker distribution by chromosomes:

```
    1     2     3     4     5     6     7     8     9    10    11    12    13    14    15    16    17 
81210 72299 59491 48601 51372 60749 47257 44078 41556 46759 50448 46061 30928 30000 28837 32101 32388 
   18    19    20    21    22    23    24    26 
24354 29060 24220 12609 15119 22112  1343   143 
```
**PLINK doesn't clean non-autosomal chromosomes**
```{r,eval=FALSE}
 table(b2miss[b2miss$F_MISS>=0.01,"CHR"])

# 23  24 
# 61 129 
```


##### 5.4 Other concerns in markers  
Markers that are in HWE are flagged but not removed from this dataset.   

##### 5.5 Summary chart of data cleaning   
![](flowchart for QC_GWAS.png)